### PR TITLE
#166789885 Pagination support for articles

### PR DESCRIPTION
**Pull request by manzif**
#### What does this PR do?

 - Pagination support for articles

#### Description of Task to be completed?

  - add a helper for article pagination
  - add test for the feature

#### How should this be manually tested?

When you access `/articles'` to get all articles, you can choose which page you want and what articles you want to read

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [#166789885](https://www.pivotaltracker.com/story/show/166789885)

## 1. Mockups

image

image

## 2. Documentation

image

image

image

#### Questions:

- N/A

### CODE REVIEW

File: swagger.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "version": "1.0.0",
       "description": "Authors Haven Documentation with Swagger"
   },
-  "host": "ah-lobos-backend-swagger.herokuapp.com",
+  "host": "localhost:7000",
```
Comment (j4l13n) [inline swagger.json]: - please put back the heroku url

Comment (manzif) [reply swagger.json]: Resolved

**Review by j4l13n — APPROVED** [review]

**Comment by adafia** [comment]
could you please change the colors of the rating stars.

**Review by adafia — APPROVED** [review]

**Review by j4l13n — APPROVED** [review]

File: src/controllers/articleController.js [hunk 2]
```diff
@@ -160,21 +161,29 @@ class ArticleManager {
...
+          error: 'there is no article'
+        });
+      }
       return res.status(200).json({
         articles: articlesList
       });
     } catch (error) {
-      return res.status(404).json({
-        error: 'there is no article'
+      return res.status(409).json({
```
Comment (Emile-Nsengimana) [inline src/controllers/articleController.js]: can you use an HTTP status code that correspond to the error you are getting?

**Review by Dom58 — COMMENTED** [review]
Change your mockup design!

File: src/controllers/articleController.js (hunk 2 above)
Comment (manzif) [reply src/controllers/articleController.js]: Resolved

**Comment by manzif** [comment]
> Change your mockup design!

Resolved

**Review by Dom58 — APPROVED** [review]

**Pull request merged**